### PR TITLE
Fully cover SystemRequest and SubscribeVehicleData request by UT

### PR DESCRIPTION
**Pull request by LevchenkoS**
Deleted incorrect tests for SystemRequest
Next test cases has been deleted:
- `SystemRequestTest.Run_FileNameContainSlash_InvalidData`
  Reason: `SystemRequest` haven't required functionality for this one
- `SystemRequestTest.Run_TypeQueryAps_UnsupportedResourse`
  Reason: `SystemRequest` support `QUERY_APPS`

-
Completed covering of `SystemRequest` command.

-
Fixed VehicleDataResultCode sending to mobile.
The problem was that we are trying to assign SmartType_Area to
a type SmartType_Map which endup in cleanup data
(VehicleInfoSubscriptions is a set of uint32_t values).

Closes bug: [APPLINK-28108](https://adc.luxoft.com/jira/browse/APPLINK-28108)

-
Completed covering of `SubscribeVehicleDataRequest` command.

-
Related to: [APPLINK-28059](https://adc.luxoft.com/jira/browse/APPLINK-28059), [APPLINK-28076](https://adc.luxoft.com/jira/browse/APPLINK-28076)

@AGaliuzov, @Kozoriz, @LuxoftAKutsan, @VProdanov, @wolfylambova22, please, review.

### CODE REVIEW

File: src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc [hunk 1]
```diff
@@ -117,6 +117,19 @@ MATCHER_P6(CheckResponseWithKeyParams,
          key_result_code == key_code;
 }
 
+class TestSubscribeVehicleDataRequest : public SubscribeVehicleDataRequest {
+ public:
+  TestSubscribeVehicleDataRequest(const MessageSharedPtr& message,
+                                  am::ApplicationManager& application_manager)
+      : SubscribeVehicleDataRequest(message, application_manager) {}
+
+  virtual ~TestSubscribeVehicleDataRequest() {}
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc]: @LevchenkoS why do you need empty virtual destructor? Does SubscribeVehicleDataRequest have not destructor?

File: src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc [hunk 2]
```diff
@@ -390,6 +403,82 @@ TEST_F(SubscribeVehicleDataRequestTest, OnEvent_AppAlreadySubscribed_Ignored) {
...
+                              kKeyName,
+                              data_type,
+                              mobile_apis::VehicleDataResultCode::VDRC_SUCCESS),
+                          _));
+  EXPECT_CALL(*app_, UpdateHash());
+  command->on_event(event);
+}
+
+TEST_F(SubscribeVehicleDataRequestTest, Run_HasDisallowedParams_DISALLOWED) {
+  (*msg_)[am::strings::params][am::strings::connection_key] = kKey;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc]: @LevchenkoS I can see that you renamed kKey to kConnectionKey. Please use this rename, it is good.

Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc]: @LuxoftAKutsan, fixed in 7f4b7df.

File: src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc (hunk 1 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc]: @LuxoftAKutsan, fixed in 7f4b7df.

File: src/components/application_manager/src/commands/mobile/subscribe_vehicle_data_request.cc [hunk 3]
```diff
@@ -363,18 +363,36 @@ void SubscribeVehicleDataRequest::AddAlreadySubscribedVI(
   using namespace mobile_apis;
   VehicleInfoSubscriptions::const_iterator it_same_app =
       vi_already_subscribed_by_this_app_.begin();
+  const VehicleData& vehicle_data = MessageHelper::vehicle_data();
+
   for (; vi_already_subscribed_by_this_app_.end() != it_same_app;
        ++it_same_app) {
-    msg_params[*it_same_app][strings::result_code] =
-        VehicleDataResultCode::VDRC_DATA_ALREADY_SUBSCRIBED;
+    for (VehicleData::const_iterator vd_it = vehicle_data.begin();
```
Comment (AGaliuzov) [inline src/components/application_manager/src/commands/mobile/subscribe_vehicle_data_request.cc]: @LevchenkoS  this cycle looks like `find` algorithm.

Could you substitute raw loop with standard algorithm?

Comment (LevchenkoS) [inline src/components/application_manager/src/commands/mobile/subscribe_vehicle_data_request.cc]: @AGaliuzov, fixed in c219a77.

**Pull request merged**